### PR TITLE
Fix polygonize with LinRange

### LINKED ISSUE #187: Polygonize example is broken
Trying out the example here: https://juliageo.org/GeometryOps.jl/stable/source/methods/polygonize#Polygonizing-raster-data

Results in an error

```julia
julia> using GeometryOps, GLMakie, GeoMakie

julia> n = 49
49

julia> xs, ys = LinRange(-3, 3, n), LinRange(-3, 3, n);

julia> zs = Makie.peaks(n);

julia> polygons = polygonize(xs, ys, 0.8 .< zs .< 3.2)
ERROR: MethodError: no method matching similar(::Type{Vector{Tuple{Float64, Float64}}}, ::LinRange{Float64, Int64})
```

### PR DESCRIPTION
**Pull request by asinghvi17**
Fix #187

### CODE REVIEW

**Comment by rafaqz** [comment]
Did you load OffsetArrays.jl and see if that still works?

**Comment by rafaqz** [comment]
I think we should get rid of `similar` and just define the undef vector

**Comment by asinghvi17** [comment]
Yes, this seems to work with OffsetArrays loaded, but I'll add a test anyway

File: src/methods/polygonize.jl [hunk 1]
```diff
@@ -121,8 +121,8 @@ function _polygonize(f::Base.Callable, xs::AbstractRange, ys::AbstractRange, A::
     ybounds = first(ys) - yhalf : step(ys) : last(ys) + yhalf
     Tx = eltype(xbounds)
     Ty = eltype(ybounds)
-    xvec = similar(Vector{Tuple{Tx,Tx}}, xs)
-    yvec = similar(Vector{Tuple{Ty,Ty}}, ys)
+    xvec = similar(Vector{Tuple{Tx,Tx}}, axes(xs))
```
Comment (rafaqz) [inline src/methods/polygonize.jl:124]:
```suggestion
    xvec = Vector{Vector{Tuple{Tx,Tx}}}(undef, length(axes(xs)))
```

Wont something like this work and skip the complexity of `similar`?

**Comment by tiemvanderdeure** [comment]
Now that https://github.com/JuliaGeo/GeometryOps.jl/pull/202 is merged we should close this

**Pull request closed without merging**